force-app/main/default/classes/SOQL.cls

```diff
@@ -271,11 +339,13 @@ public with sharing class SOQL {
      */
     public static List<Contact> getAccountsWithContactsByState(Id accountId) {
         // Query the Account
-
+        List<Account> acc = [SELECT Id, BillingState FROM Account WHERE Id = :accountId];
         // Store the Account's BillingState
 
         // Query the Contacts
-        return null; // Replace null with the variable you used to store the result
+        string billstate = acc[0].BillingState;
+        List<Contact> con = [SELECT Id, MailingState FROM Contact WHERE MailingState = :billstate];
+        return con; // Replace null with the variable you used to store the result
```

Comment:
We did not go over this, but can you do this with just 1 query? HINT: there will be a query in your criteria